eo-maven-plugin/src/test/java/org/eolang/maven/VerifyMojoTest.java

```diff
@@ -115,4 +119,77 @@ void detectsCriticalError(@TempDir final Path temp) {
             "Program with duplicate names should have failed on critical error, but it didn't"
         );
     }
+
+    @Test
+    void failsOptimization(@TempDir final Path temp) {
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> new FakeMaven(temp)
+                .withProgram(
+                    "+package f",
+                    "+alias THIS-IS-WRONG org.eolang.io.stdout\n",
+                    "[args] > main",
+                    "  (stdout \"Hello!\").print > @"
+                )
+                .execute(new FakeMaven.Verify())
+        );
+    }
+
+    @Test
+    void failsOnWarning(@TempDir final Path temp) throws Exception {
+        final FakeMaven maven = new FakeMaven(temp)
+            .withProgram(
+                "+architect yegor256@gmail.com",
+                "+tests",
+                "+package org.eolang.examples\n",
+                "[] > main",
+                "  [] > @",
+                "    hello > test"
+            );
+        this.applyXsl(
+            "org/eolang/maven/set-warning-severity.xsl",
+            maven.execute(ParseMojo.class)
+                .result()
+                .get("target/1-parse/foo/x/main.xmir")
+        );
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> maven.with("failOnError", false)
+                .with("failOnWarning", true)
+                .execute(VerifyMojo.class)
+        );
+    }
+
+    @Test
+    void failsOnCritical(@TempDir final Path temp) {
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> new FakeMaven(temp)
+                .withProgram(
+                    "+package f\n",
+                    "[args] > main",
+                    "  seq > @",
+                    "    TRUE > x",
+                    "    FALSE > x"
+                )
+                .execute(new FakeMaven.Verify())
+        );
+    }
+
+    /**
+     * Apply XSL transformation.
+     * @param xsl Path to XSL within classpath
+     * @param xml Path to XML to be tranformed
+     */
+    private void applyXsl(final String xsl, final Path xml) throws Exception {
```

Comment:
@levBagryansky can we make this method static?

Reply:
@maxonfjvipon done.